src-electron/db/query-util.js

```diff
@@ -83,5 +83,71 @@ function sqlQueryForDataTypeByNameAndClusterId(
   return clusterId ? queryWithClusterId : queryWithoutClusterId
 }
 
+/**
+ * Formulate a sqlite query string for a data type from the given cluster name and package IDs.
+ * @param {*} typeDiscriminator
+ * @param {*} name data type name
+ * @param {*} clusterName
+ * @param {*} packageIds
+ * @param {*} options
+ * @returns SQLite query string
+ */
+function sqlQueryForDataTypeByNameAndClusterName(
+  typeDiscriminator,
+  name,
+  clusterName,
+  packageIds,
+  options = {}
+) {
+  let typeTableName = typeDiscriminator.toUpperCase()
+  let numberExtensionString =
+    typeDiscriminator == 'number' ? 'NUMBER.IS_SIGNED, ' : ''
+  let checkLowerCaseString =
+    typeDiscriminator != 'number' && typeDiscriminator != 'struct'
+      ? `OR DATA_TYPE.NAME = '${name}'`
+      : ''
+  let structExtensionString =
+    typeDiscriminator == 'struct'
+      ? 'STRUCT.IS_FABRIC_SCOPED, DATA_TYPE.DISCRIMINATOR_REF, '
+      : ''
+  let selectQueryString = `
+  SELECT
+    ${typeTableName}.${typeTableName}_ID,
+    ${structExtensionString}
+    DATA_TYPE.NAME AS NAME,
+    ${numberExtensionString}
+    ${typeTableName}.SIZE AS SIZE,
+    CLUSTER.NAME AS CLUSTER_NAME,
+    REPLACE(CLUSTER.NAME, ' ', '') AS CLUSTER_NAME_WITHOUT_SPACES
+  FROM ${typeTableName}
+  INNER JOIN
+    DATA_TYPE
+  ON
+    ${typeTableName}.${typeTableName}_ID = DATA_TYPE.DATA_TYPE_ID
+  LEFT JOIN
+    DATA_TYPE_CLUSTER
+  ON
+    DATA_TYPE_CLUSTER.DATA_TYPE_REF = ${typeTableName}.${typeTableName}_ID 
+  LEFT JOIN
+    CLUSTER
+  ON
+    DATA_TYPE_CLUSTER.CLUSTER_REF = CLUSTER.CLUSTER_ID
+  WHERE
+    (DATA_TYPE.NAME = '${name}' ${checkLowerCaseString})
+  AND
+    DATA_TYPE.PACKAGE_REF IN (${dbApi.toInClause(packageIds)}) `
+
+  let whereClause = !options.ignoreClusterWhereClause
+    ? clusterName
+      ? `AND (CLUSTER.NAME = '${clusterName}' OR CLUSTER_NAME_WITHOUT_SPACES = '${clusterName}')`
```

Comment:
After https://github.com/project-chip/connectedhomeip/pull/38267 lands I don't think you need this bit at all.

Reply:
My suggestion is to still keep it if someone loads the chip repo before your change is added with latest ZAP it still gives a reasonable output.

Reply:
I guess it may not happen because the corresponding xml changes don't exist yet. However, should we still keep it?
I can remove it if we definitely don't need that safety because the use case won't exist.

Reply:
That use case won't exist, I expect, so let's just take this bit out.